### DISCUSSION TITLE
Test State Scope

### DISCUSSION THREAD

**Discussion by evilrobot-01** [foundry-rs/foundry#1241] [Q&A]
New to Solidity, but loving Foundry so far! 

I just wanted to ask about the scope of state across tests/runs. Does each test have its own isolated state within each test run? I am not sure whether it is better to construct the instance of the contract under test as a state variable on the test contract or within the test contract's setUp() function, or whether it even matters.

**Comment by onbjerg** [comment]
Yes, each test has its own isolated state. `setUp` is like `beforeEach` in Hardhat/Truffle and there is no way to share state between tests :)
